tests/addEmailIdentityWorkflow.test.tsx

```diff
@@ -0,0 +1,77 @@
+import { enableFetchMocks } from "jest-fetch-mock";
+import React from "react";
+
+import { IdentityTypes } from "../src/@types/Identity";
+import { ProviderUser } from "../src/@types/ProviderUser";
+import Index from "../src/pages/index";
+import { render, fireEvent } from "./config/test-utils";
+
+enableFetchMocks();
+
+describe("addEmailIdentityWorkflow", () => {
+  type MockedFetchedData = {
+    data: {
+      provider: ProviderUser;
+    };
+  };
+
+  const mockedFetchedData: MockedFetchedData = {
+    data: {
+      provider: {
+        id: "4a5f8589-0d91-4a69-924a-6f227a69666d",
+        name: "Mocked Provider",
+        user: {
+          id: "299d268e-3e19-4486-9be7-29c539d241ac",
+          identities: [
+            {
+              id: "7f8d168a-3f65-4636-9acb-7720a212680e",
+              primary: true,
+              status: "validated",
+              type: IdentityTypes.PHONE,
+              value: "0123456789",
+            },
+            {
+              id: "8f79dcc1-530b-4982-878d-33f0def6a7cf",
+              primary: true,
+              status: "validated",
+              type: IdentityTypes.EMAIL,
+              value: "test@test.test",
+            },
+          ],
+        },
+      },
+    },
+  };
+
+  test("Add email identity user flow with button click", async () => {
+    const { getByText, getByRole, getByTestId } = render(
+      <Index fetchedData={mockedFetchedData} />,
+      {},
+    );
+
+    const addAnEmailButton = getByText("Add an email");
+
+    fireEvent.click(addAnEmailButton);
+
+    const identityForm = getByTestId("identity-form");
```

Comment:
Not a blocking comment but are testid required? Could we envision another way or is it *really* simpler?

Reply:
The only other option is to use something like `querySelector`, but we lose the goodies from `testing-library` if we do this. I really don't like the idea of changing components to enable testing like this, but that's the cleaner way I found.

For some reason, `form` is not a role so I can't select it with the lib. Maybe I could do it with catching the main content a searching for the child, or something along those line, but it would add a lot of bloat, and I wanted to go "fast".

Reply:
ok, well let's go with that at first then